packages/next/src/client/app-dir/link.react-server.tsx

```diff
@@ -0,0 +1,31 @@
+import type { ComponentProps } from 'react'
+import ClientLinkComponent, { type LinkProps, useLinkStatus } from './link'
+
+export default function LinkComponent(
+  props: ComponentProps<typeof ClientLinkComponent>
+) {
+  const isLegacyBehavior = props.legacyBehavior
+  const childIsHostComponent =
+    typeof props.children === 'string' ||
+    typeof props.children === 'number' ||
+    typeof (props.children as any)?.type === 'string'
+  const childIsClientComponent =
+    (props.children as any)?.type?.$$typeof ===
+    Symbol.for('react.client.reference')
+
+  if (isLegacyBehavior && !childIsHostComponent && !childIsClientComponent) {
+    if ((props.children as any)?.type.$$typeof === Symbol.for('react.lazy')) {
+      console.error(
+        `Using a Lazy Component as a direct child of a Server Component that renders \`<Link legacyBehavior>\` is not supported. If you need legacyBehavior, wrap your Lazy Component in a Client Component that renders the Link's \`<a>\` tag.`
```

Comment:
I think this phrasing is wrong. Or at least it doesn't read the way I think it is supposed to.
```suggestion
        `Using a Lazy Component as a direct child of \`<Link legacyBehavior>\` from a Server Component is not supported. If you need legacyBehavior, wrap your Lazy Component in a Client Component that renders the Link's \`<a>\` tag.`
```

It's a mouthful. We can add an explainer page later if we want before final release